openedx/openedx-platform
# Changing the functionality of "View live" button  on CMS

# Change "View Live" Button Functionality in CMS

 Staff Member should have the ablity to view the the course from CMS "View Live" option on Multiple sites, If the course is shared on multiple sites.
 
## Status

* [ ] Idea
* [ ] In development
* [ ] Won't do


## Migrations

No


## Description

We have support for multiple orgs in sites which provides the users to publish the same course in Multiple Sites.
Staff Member should have the ablity to view the course in LMS before publishing the course in any site.
In current implementation, we are able to see the vewlive only on one site.


## Related PRs

[WL-926 Add support for multi-org sites #14471](https://github.com/edx/edx-platform/pull/14471)


List related PRs:

| branch       | PR                                             |
| ------------ | ---------------------------------------------- |
| edx-platform | https://github.com/edx/edx-platform/pull/14471 |


## Todos

Do you think edX will need to perform the following?

* [ ] Product manager review
* [X] UX/UI review
* [ ] Technical design review


## Steps to Test or Reproduce

1. create a course to publish the course in multiple site.
2. click on "view live" button on CMS
3. we can see the course on only one site but we should have the ability to view in multiple sites.


## Proposed Solution
we can change the "view live" from hyperlink to drop-down. the drop-down list will dynamically get the list of sites for each course based on the organization. 
For example,
1) if  a course was shared between **two** sites with same org then the "view live" drop-down for that course will have two site names listed on it.
2)  if  a course was shared between **three** sites with same org then the "view live" drop-down for that course will have Three site names listed on it.
